### PR TITLE
fix(sql-check): cannot get oracle affect sql rows

### LINKED ISSUE #3789: [Bug]: The insert statement does not take effect in Limit the number of lines affected by SQL
### ODC version

4.3.3

### OB version

Native Oracle

### What happened?

Set the number of affected rows to 0, select oracle, select Require approval, and execute an insert statement, in which case the insert statement is executed normally
image

### What did you expect to happen?

Trigger the approval process

### How can we reproduce it (as minimally and precisely as possible)?

Set the number of affected rows to 0, select oracle, select Require approval, and execute an insert statement, in which case the insert statement is executed normally

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by zijiacj**
#### Which issue(s) this PR fixes:

Fixes #3789

#### Additional documentation e.g., usage docs, etc.:

```docs

```

### CODE REVIEW

File: server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java [hunk 1]
```diff
@@ -115,7 +115,7 @@ private long getOracleAffectedRows(String originalSql, JdbcOperations jdbcOperat
                 "SELECT * FROM TABLE(DBMS_XPLAN.DISPLAY('PLAN_TABLE', '" + ODC_TEMP_EXPLAIN_STATEMENT_ID + "', 'ALL'))";
         List<String> queryResults = jdbcOperations.query(getPlanSql, (rs, rowNum) -> rs.getString("PLAN_TABLE_OUTPUT"));
         long estRowsValue = 0;
-        for (int rowNum = 5; rowNum < queryResults.size(); rowNum++) {
+        for (int rowNum = 2; rowNum < queryResults.size(); rowNum++) {
```
Comment (yhilmare) [inline server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: why 2?

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: image
4 or before 4 is fine

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: why do you set 5 before? does ob-mysql has the same problems?

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: no

Comment (zijiacj) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: image
The index will not get the value of column row from index 5

Comment (yhilmare) [reply server/odc-service/src/main/java/com/oceanbase/odc/service/sqlcheck/rule/OracleAffectedRowsExceedLimit.java:118]: from row 0 may be better?

**Review by yhilmare — APPROVED** [review]
lgtm

**Pull request merged**